src/PhpParser/Printer/BetterStandardPrinter.php

```diff
@@ -38,6 +46,14 @@ public function __construct(array $options = [])
         $this->insertionMap['Expr_Closure->returnType'] = [')', false, ': ', null];
     }
 
+    public function printFormatPreserving(array $stmts, array $origStmts, array $origTokens): string
+    {
+        // detect per print
+        $this->detectTabOrSpaceIndentCharacter($stmts);
```

Comment:
@dereuromark Here is the important change. PhpParser prints everything with spaces-only by default. So there was actually 1 indent in the code, just not 1 tab, but 1 space.

Detecting spaces per line would be slow and not very useful, so 1st newlinew indent kind is taken into account - see `detectTabOrSpaceIndentCharacter()` bellow.

What do you think?

Reply:
Sounds reasonable to me :)

Reply:
Ok :), let's go with then. If anything breaks in the future, we know wher to look for.